### PR TITLE
Adding date fns and removing momentjs libs. Some proposed approaches.

### PR DESCRIPTION
**Pull request by stratease**
### 🎫 Ticket

[TEC-4242]

### 🗒️ Description

Updating our javascript modules to leverage `date-fns` and vanilla JS instead of `moment`. This update touches Block Editor and Classic Editor differently.

### 🎥 Artifacts 

image

### ✔️ Checklist
- [ ] Changelog entry in the `readme.txt` file.
- [ ] Code is covered by **NEW** `wpunit` or `integration` tests.
- [ ] Code is covered by **EXISTING** `wpunit` or `integration` tests.
- [ ] Are all the **required** tests passing?
- [ ] Automated code review comments are addressed.
- [ ] Have you added Artifacts?
- [ ] Check the base branch for your PR.
- [ ] Add your PR to the project board for the release.

[TEC-4242]: https://stellarwp.atlassian.net/browse/TEC-4242?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: jest.setup.js [hunk 1]
```diff
@@ -27,4 +27,3 @@ global.render = render;
 global.mount = mount;
 global.renderer = renderer;
 
-moment.tz.setDefault( 'UTC' );
```
Comment (stratease) [inline jest.setup.js:30]: Will we need this? Global settings?

File: src/modules/elements/time-picker/element.js [hunk 2]
```diff
@@ -89,9 +87,9 @@ const TimePicker = ( {
 	);
 
 	const formatLabel = ( seconds ) => {
-		return momentUtil
+		return dateUtil
```
Comment (stratease) [inline src/modules/elements/time-picker/element.js:90]:
```suggestion
		return dateUtil
```
Maybe we abstract out whatever we use inside our "util". This avoids being tied to a particular lib, centralizing a dependency?

**Review by stratease — COMMENTED** [review]
Just a thought, but if we migrate directly "inline" it could be quite cumbersome to do. I don't recommend we clone `moments` api, but perhaps we abstract out a bit and hide the implementation details. 

For example: 

Currently it's API is like this:
```
var comparison_end_time = moment( obj.end_time, 'h:mm a' ).format( 'HH:mm' );
```
And if we directly move to `date-fns` it would be something like this:
```
var comparison_start_time = obj.start_time ? dateFns.format( dateFns.toDate( obj.start_date + ' ' + obj.start_time ), 'HH:mm' ) : null;
```
with potentially more refactoring to be done to handle a native `Date` vs `moment`. 

Perhaps we middle ground, and use `date-fns` and `vanilla` inside a util that will hide the implementation details.

I'm not sold on either way, both have pros and cons.

**Review by bordoni — CHANGES_REQUESTED** [review]
We need to update the `package-safelist.json`

**Comment by Camwyn** [comment]
We wound up going with Dayjs for now

**Pull request closed without merging**